host/src/c_api/c_api_mc.cc

```diff
@@ -840,30 +831,30 @@ XGB_DLL int encrypt_data_with_pk(char* data, size_t len, uint8_t* pem_key, size_
       (const unsigned char*) data,
       (unsigned char*) encrypted_data);
   if (res != 0) {
-    std::cout << "mbedtls_rsa_pkcs1_encrypt failed\n";
     mbedtls_pk_free(&key);
-    return res;
+    LOG(FATAL) << "mbedtls_rsa_pkcs1_encrypt failed.";
   }
 
   *encrypted_data_size = mbedtls_pk_rsa(key)->len;
 
   mbedtls_pk_free( &m_pk_context );
   mbedtls_ctr_drbg_free( &m_ctr_drbg_context );
   mbedtls_entropy_free( &m_entropy_context );
-  return 0;
+  API_END();
 }
 
 XGB_DLL int sign_data_with_keyfile(char *keyfile, uint8_t* data, size_t data_size, uint8_t* signature, size_t* sig_len) {
+  API_BEGIN();
   mbedtls_pk_context pk;
   mbedtls_pk_init( &pk );
 
   int ret;
   if((ret = mbedtls_pk_parse_keyfile( &pk, keyfile, "")) != 0) {
-    LOG(FATAL) <<"signing failed -- mbedtls_pk_parse_public_keyfile returned" << ret;
+    LOG(FATAL) << "signing failed -- mbedtls_pk_parse_public_keyfile returned " << ret;
   }
 
   ret = sign_data(pk, data, data_size, signature, sig_len);
-  return ret;
+  API_END();
```

Comment:
Can you also add this to any other functions that can be called by the client? E.g. `encrypt_file_with_keybuf` and `decrypt_file_with_keybuf`

Reply:
Oops I must've missed these. Fixed